src/features/PassSurvey/model/ConditionalLogicBuilder.ts

```diff
@@ -7,22 +7,32 @@ type Item = appletModel.ItemRecord;
 
 export type ItemMapByName = Record<string, Item>;
 
-class ConditionalLogicBuilder {
+type ProcessResult = {
+  visibleItems: Item[];
+  hiddenItemIds: Set<string>;
+};
+
+interface IConditionalLogicFilter {
```

Comment:
I suggest renaming it to `IFilter`and moving to the shared level

Reply:
Done